client/client_pool/src/external_client.cpp

```diff
@@ -112,6 +117,9 @@ void ConcordClient::AddPendingRequest(std::vector<uint8_t>&& request,
 
   bftEngine::ClientReply pending_reply;
   if (reply_size) {
+    // We assume a callback is provided in order to distinguish later whether the replyBuffer should be used
+    // TODO: Do we need the else-path?
+    ConcordAssert(callback != nullptr);
```

Comment:
This code works now without CB provided and reply_size set to max_reply_size_.

Reply:
We clarified: We have tests that don't provide a callback but valid reply buffers. So we cannot have an assert here.